react/finger-exercises/redux/02 - react-redux/src/app/screens/App/index.js

```diff
@@ -1,53 +1,39 @@
 import React, { Component, Fragment } from 'react';
-import store from '@redux/store';
+import { connect } from 'react-redux';
+import { arrayOf, func, string } from 'prop-types';
 import Navbar from '@components/Navbar';
 import Footer from '@components/Footer';
 
+import actionsCreators from '../../../redux/book/actions';
+
 import Book from './components/Book';
 import Search from './components/Search';
 import ShoppingCart from './components/ShoppingCart';
 import styles from './styles.scss';
 
 class App extends Component {
-  state = {
-    books: [],
-    bookSelected: []
-  };
-
   componentDidMount() {
-    store.subscribe(() => {
-      const { books, bookSelected } = store.getState();
-      this.setState({ books, bookSelected });
-    });
-    // TODO to implement the dispatch
+    this.props.getBooks();
   }
 
-  // TODO to implement the dispatch
-  onSearch = value => {};
-
-  // TODO to implement the dispatch
-  addToCart = item => {};
+  onSearch = value => this.props.searchBook(value);
```

Comment:
What about removing the `onSearch` method and just passing `searchBook` to `<Search onSearch={ ... } />`?
As it is this method is just bypassing the `value`, there is no extra logic..